packages/flutter/lib/src/material/page_transitions_theme.dart

```diff
@@ -629,3 +738,193 @@ class PageTransitionsTheme with Diagnosticable {
     );
   }
 }
+
+// Take an image and draw it centered and scaled. The image is already scaled by the [pixelRatio].
+void _drawImageScaledAndCentered(PaintingContext context, ui.Image image, double scale, double opacity, double pixelRatio) {
+  if (scale <= 0.0 || opacity <= 0.0) {
+    return;
+  }
+  final Paint paint = Paint()
+    ..filterQuality = ui.FilterQuality.low
+    ..color = Color.fromRGBO(0, 0, 0, opacity);
+  final double logicalWidth = image.width / pixelRatio;
+  final double logicalHeight = image.height / pixelRatio;
+  final double scaledLogicalWidth = logicalWidth * scale;
+  final double scaledLogicalHeight = logicalHeight * scale;
+  final double left = (logicalWidth - scaledLogicalWidth) / 2;
+  final double top = (logicalHeight - scaledLogicalHeight) / 2;
+  final Rect dst = Rect.fromLTWH(left, top, scaledLogicalWidth, scaledLogicalHeight);
+  context.canvas.drawImageRect(image, Rect.fromLTWH(0, 0, image.width.toDouble(), image.height.toDouble()), dst, paint);
+}
+
+void _updateScaledTransform(Matrix4 transform, double scale, Size size) {
+  transform.setIdentity();
+  if (scale == 1.0) {
+    return;
+  }
+  transform.scale(scale, scale);
+  final double dx = ((size.width * scale) - size.width) / 2;
+  final double dy = ((size.height * scale) - size.height) / 2;
+  transform.translate(-dx, -dy);
+}
+
+mixin _ZoomTransitionBase {
+  // Don't rasterize if:
+  // 1. Rasterization is disabled by the platform.
+  // 2. The animation is paused/stopped.
+  // 3. The values of the scale/fade transition do not
+  //    benefit from rasterization.
+  final RasterWidgetController controller = RasterWidgetController();
+
+  late Animation<double> fadeTransition;
+  late Animation<double> scaleTransition;
+
+  bool get allowRasterization;
```

Comment:
nit: the mixin doesn't care about this setting anymore. Maybe remove it here and just make the overrides private in the subclasses?